src/components/ListItem.jsx

```diff
@@ -1,5 +1,56 @@
 import './ListItem.css';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+const currentTime = new Date();
+
+export function ListItem({ name, items, listToken }) {
+	const { isChecked, id, totalPurchases, dateLastPurchased } = items;
+
+	const currentDate = new Date();
+
+	if (isChecked) {
+		const currentTimeToSec = currentDate.getTime() / 1000;
+		const lastPurchaseToMilSec = dateLastPurchased.seconds * 1000;
+		const timeElapsed = currentTimeToSec - lastPurchaseToMilSec;
+
+		const timeToUncheck = 86400;
+		if (timeElapsed >= timeToUncheck) {
+			const itemData = {
+				isChecked: false,
+			};
+			updateItem(listToken, id, itemData);
+		}
+	}
+
+	const handleCheckbox = () => {
+		if (isChecked) {
+			const itemData = {
+				isChecked: false,
+			};
+			updateItem(listToken, id, itemData);
+		} else {
+			const updateTotalPurchases = totalPurchases + 1;
```

Comment:
The name `updateTotalPurchases` feels like a function name instead of a variable. It would be better if the variable was named `updatedTotalPurchases` :slightly_smiling_face:

Reply:
We misspelled it :see_no_evil: